server/src/main/java/com/linecorp/centraldogma/server/internal/api/converter/WatchRequestConverter.java

```diff
@@ -55,6 +55,8 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
             } else {
                 timeoutMillis = DEFAULT_TIMEOUT_MILLIS;
             }
+            // Update timeout according to the watch API specifications.
+            ctx.setRequestTimeoutMillis(timeoutMillis);
```

Comment:
- How about using `LongMath.saturatedAdd(timeoutMillis, ctx.requestTimeoutMillis())` ?
- How about enforcing upper bound to the timeout? We don't want to respect a client that specifies a timeout longer than 1 year. You might also want to update `CentralDogmaTimeoutScheduler` as well for consistent behavior.

Reply:
Added [WatchTimeout](https://github.com/line/centraldogma/pull/325/files#diff-cb3f0f652cf2ec37af3f5db13a135b0d) class. Please take a look again. :-)